src/pybind/mgr/orchestrator/module.py

```diff
@@ -788,17 +788,42 @@ def _daemon_add_osd(self,
         usage = """
 Usage:
   ceph orch daemon add osd host:device1,device2,...
+  ceph orch daemon add osd host:block=device1,device2,db=device3,osds_per_device=2,...
 """
         if not svc_arg:
             return HandleCommandResult(-errno.EINVAL, stderr=usage)
         try:
-            host_name, block_device = svc_arg.split(":")
-            block_devices = block_device.split(',')
-            devs = DeviceSelection(paths=block_devices)
+            host_name, raw = svc_arg.split(":")
+            drive_group_spec = {
+                'data_devices': []
+            }  # type: Dict
+            drv_grp_spec_arg = None
+            values = raw.split(',')
+            while values:
+                v = values[0].split(',', 1)[0]
+                if '=' in v:
+                    drv_grp_spec_arg, value = v.split('=')
+                    if drv_grp_spec_arg in ['data_devices',
+                                            'db_devices',
+                                            'wal_devices',
+                                            'journal_devices']:
+                        drive_group_spec[drv_grp_spec_arg] = []
+                        drive_group_spec[drv_grp_spec_arg].append(value)
```

Comment:
I see here we're checking specifically for `data_devices=`, `db_devices=` etc. but the example line added in the "Usage" comment at the top uses `block=`, `db=`. Does that case still work even though we'll never do something like `drive_group_spec["block"] = []`?

Reply:
Also, are we interested in adding `extra_container_args`, `data_directories` to this list so they can be passed in this command (as they are lists) or is that too complex for something on the command line

Reply:
> I see here we're checking specifically for `data_devices=`, `db_devices=` etc. but the example line added in the "Usage" comment at the top uses `block=`, `db=`. Does that case still work even though we'll never do something like `drive_group_spec["block"] = []`?

it's a typo, need to be modified 🙂

Reply:
> Also, are we interested in adding `extra_container_args`, `data_directories` to this list so they can be passed in this command (as they are lists) or is that too complex for something on the command line

I don't know what it represents, but at first glance, I don't think that should be too complex, we can add them IMO. I just need to test I guess